packages/core/src/interpreter.ts

```diff
@@ -50,29 +45,29 @@ export function createInterpreter<T extends AnyInterpreter, U extends {} = {}>(
 
   switch (options ? options.numberOfArguments : 0) {
```

Comment:
I also don't think you have to make this differentiation by the number of arguments, whith my approach it works with arbitairy numer of arguments. 
So an interpreter can have any amount of arguments and the context is still passed as last argument.
```ts
const interpret = ((condition : Condition, ...params: any[]) => {
    const interpreterName = getInterpreterName(condition, options);
    const interpretOperator = getInterpreter(interpreters, interpreterName);
    return interpretOperator.apply(null, [condition, ...params, defaultContext]);// eslint-disable-line @typescript-eslint/no-use-before-define
  }) as unknown as InterpretationContext<T>['interpret'];
```